### PR TITLE
[NC-2044] jsonrpc authentication cli options & acceptance tests

### PR DESCRIPTION
**Pull request by Errorific**
Adds the cli options to turn on authentication and provide a credential source to the toml authentication backend. Also adds random generation of JWT certificates. Acceptance tests wired in for logging in.

### CODE REVIEW

File: acceptance-tests/src/test/java/tech/pegasys/pantheon/tests/acceptance/dsl/httptransaction/HttpRequestFactory.java [hunk 1]
```diff
@@ -0,0 +1,69 @@
+/*
```
Comment (macfarla) [inline acceptance-tests/src/test/java/tech/pegasys/pantheon/tests/acceptance/dsl/httptransaction/HttpRequestFactory.java]: nit: package name should be httpTransaction

Comment (macfarla) [reply acceptance-tests/src/test/java/tech/pegasys/pantheon/tests/acceptance/dsl/httptransaction/HttpRequestFactory.java]: or maybe just http?

Comment (Errorific) [reply acceptance-tests/src/test/java/tech/pegasys/pantheon/tests/acceptance/dsl/httptransaction/HttpRequestFactory.java]: I looked through the codebase and couldn't find examples of mixed case package names, found the opposite with all lower case. The transaction part is being copied from tech.pegasys.pantheon.tests.acceptance.dsl.transaction as this is a different form of the same thing.

Comment (macfarla) [reply acceptance-tests/src/test/java/tech/pegasys/pantheon/tests/acceptance/dsl/httptransaction/HttpRequestFactory.java]: 👍

**Pull request merged**